benches/swap_benchmarks.rs

```diff
@@ -0,0 +1,268 @@
+        .skip_state_decode_failures(true)
+        .build()
+        .await
+        .map_err(|e| {
+            warn!("Failed to build protocol stream: {:?}", e);
+            e
+        })
+        .expect("Failed to build protocol stream")
+        .boxed();
+
+    // Load initial data and organize by protocol
+    let mut protocol_data: BenchmarkDataByProtocol = HashMap::new();
+
+    match stream.next().await {
+        Some(Ok(message)) => {
+            info!(
+                "Successfully loaded {} pairs and {} states",
+                message.new_pairs.len(),
+                message.states.len()
+            );
+            // Initialize protocol data structures
+            let protocols =
+                ["uniswap_v2", "uniswap_v3", "uniswap_v4", "balancer_v2", "curve", "ekubo_v2"];
+            for protocol in protocols {
+                protocol_data.insert(
+                    protocol.to_string(),
+                    ProtocolBenchmarkData {
+                        pools: HashMap::new(),
+                        components: HashMap::new(),
+                        states: HashMap::new(),
+                    },
+                );
+            }
+
+            // Organize pools and components by protocol
+            for (id, component) in message.new_pairs.iter() {
+                let protocol_name = map_protocol_system_to_protocol(&component.protocol_system);
+                if let Some(protocol_data) = protocol_data.get_mut(&protocol_name) {
+                    protocol_data
+                        .pools
+                        .insert(id.clone(), component.tokens.clone());
+                    protocol_data
+                        .components
+                        .insert(id.clone(), component.clone());
+                }
+            }
+
+            // Organize states by protocol using the component information
+            for (id, state) in message.states.into_iter() {
+                // Find which protocol this state belongs to by looking up the component
+                for (_protocol_name, data) in protocol_data.iter_mut() {
+                    if data.components.contains_key(&id) {
+                        data.states.insert(id.clone(), state);
+                        break;
+                    }
+                }
+            }
+        }
+        Some(Err(e)) => {
+            warn!("Error loading protocol data: {:?}", e);
+        }
+        None => {
+            warn!("No data received from protocol stream");
+        }
+    }
+
+    protocol_data
+}
+
+fn map_protocol_system_to_protocol(protocol_system: &str) -> String {
+    match protocol_system {
+        "uniswap_v2" => "uniswap_v2".to_string(),
+        "uniswap_v3" => "uniswap_v3".to_string(),
+        "uniswap_v4" => "uniswap_v4".to_string(),
+        "vm:balancer_v2" => "balancer_v2".to_string(),
+        "vm:curve" => "curve".to_string(),
+        "ekubo_v2" => "ekubo_v2".to_string(),
+        _ => protocol_system.to_string(),
```

Comment:
why is this necessary?? couldn't we just use the same terminology? 🤔

Reply:
It's not necessary it just makes the output a bit nicer is this something that really bothers you?

Reply:
for me it's unnecessary code